hw2/task04/hw2_task04.py

```diff
@@ -0,0 +1,35 @@
+"""Write a function that accepts another function as an argument.
+
+Then it should return such a function, so the every call to initial one
+should be cached.
+
+def function(a, b):
+    return (a ** b) ** 2
+
+cache_func = cache(function)
+
+some = 100, 200
+
+val_1 = cache_func(*some)
+val_2 = cache_func(*some)
+
+assert val_1 is val_2
+"""
+from typing import Callable
+
+
+def function(a: int, b: int) -> int:
+    return (a ** b) ** 2
+
+
+def cache(func: Callable) -> Callable:
+    f_cache = {}
+
+    def wrapper(*args: int) -> int:
```

Comment:
Why it is int?

Reply:
To match data, that will be given to ''function". But cache could be used for other functions (like sum of strings) too, so I changed it to Any.